tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -0,0 +1,435 @@
+          comments :+= "Not all cores in the machine are being used. " +
+            "It is recommended to use different machine."
+        }
+      }
+
+      bestConfig.set_executor_cores(numCores)
+
+      // Recommendation for 'spark.executor.memory' based on system memory, cluster scheduler
+      // and num of gpus
+      val sparkMaster = getSparkProperty(app, "spark.master")
+      val systemMemoryNum: Long = convertFromHumanReadableSize(systemProps.memory)
+      val extraSystemMemoryNum: Long = convertFromHumanReadableSize(MAX_EXTRA_SYSTEM_MEMORY)
+      val effectiveSystemMemoryNum: Long =
+        if (sparkMaster.contains("yarn") || sparkMaster.contains("k8s")) {
+          systemMemoryNum -  extraSystemMemoryNum -
+            convertFromHumanReadableSize(MIN_MEMORY_OVERHEAD)
+        } else systemMemoryNum - extraSystemMemoryNum
+      val maxExecutorMemNum: Long = convertFromHumanReadableSize(MAX_EXECUTOR_MEMORY)
+
+      val executorMemory: Long = if (systemProps.gpu_props != null) {
+        Math.min(effectiveSystemMemoryNum * 1.0 / systemProps.gpu_props.count,
+          maxExecutorMemNum).toLong
+      } else {
+        Math.min(effectiveSystemMemoryNum * 1.0 / numCores,
+          maxExecutorMemNum).toLong
+      }
+
+      if(executorMemory < convertFromHumanReadableSize(MIN_EXECUTOR_MEMORY)) {
+        comments :+= s"Executor memory is very low. " +
+          s"It is recommended to have at least $MIN_EXECUTOR_MEMORY"
+      }
+
+      bestConfig.set_executor_memory(convertToHumanReadableSize(executorMemory))
+
+      // Recommendation for 'spark.sql.shuffle.partitions' based on spill size
+      var shufflePartitions: Int = getSparkProperty(app, "spark.sql.shuffle.partitions")
+        .getOrElse("200").toInt
+
+      // TODO: Need to look at other metrics for GPU spills (DEBUG mode), and batch sizes metric
+      val totalSpilledMetrics = app.sqlTaskAggMetrics.map {
+        task => task.diskBytesSpilledSum + task.memoryBytesSpilledSum
+      }.sum
+      if (totalSpilledMetrics > 0) {
+        shufflePartitions *= DEFAULT_SHUFFLE_PARTITION_MULTIPLIER
+        // Could be memory instead of partitions
+        comments :+= "\"spark.sql.shuffle.partitions\" should be increased since spilling occurred."
+      }
+      bestConfig.set_shuffle_partitions(shufflePartitions)
+
+      // Recommendation for 'spark.sql.files.maxPartitionBytes' based on input size for each task
+      getSparkProperty(app, "spark.sql.files.maxPartitionBytes") match {
+        case None => bestConfig.set_max_partition_bytes(DEFAULT_MAX_PARTITION_BYTES)
+        case Some(maxPartitionBytes) =>
+          val taskInputSize =
+            app.sqlTaskAggMetrics.map(_.inputBytesReadAvg).sum / app.sqlTaskAggMetrics.size
+          val maxPartitionBytesNum = convertFromHumanReadableSize(maxPartitionBytes)
+          val newMaxPartitionBytes =
+            if (taskInputSize < convertFromHumanReadableSize(MIN_PARTITION_BYTES_RANGE)) {
```

Comment:
there is a lot of converting from human readable in here, if we create these global vars it might be better to use the non-human readable until we go to actually print something to the user, or have getters that do it

Reply:
added followup #6310 for this